### DISCUSSION TITLE
Molten commands VS. Quarto commands - Does quarto directly replace molten's cell commands?

### DISCUSSION THREAD

**Discussion by Or1g3n** [benlubas/molten-nvim#263] [Q&A] [answered]
Hello, I am looking to take my molten setup to the next level and have reviewed your Notebook-Setup document for your suggested setup.

I have a few questions regarding some of the commands that appear to directly overlap with those already part of molten:
1. Per the doc, "When you configure quarto, you gain access to these functions which should be mapped to commands:...". Do these new cell commands from Quarto "replace" the already existing cell commands from Molten. In other words, if I configure the Quarto plugin, does it make sense to keep my existing molten keymaps related to cells or replace them with the new Quarto ones. I would rather minimize on redundancy if the end result of the "overlapping" keymaps are the same.
2. This next question may be out of scope for you, if so, ignore. I already have LSP/Autocomplete features working with hover, go-to-definition, etc. Do I actually need to configure this in Quarto with additional keymaps? Are these Quarto LSP configurations specific to Quarto only or is the plugin able to leverage my existing LSP/autocomplete config?

Appreciate your time and thoughts!

**Comment by benlubas** [comment, marked as answer]
1. quarto and molten "cells" are different. Quarto knows about the markdown code "cell", while molten only knows "this range of lines that the user told me to run is a cell". So if you use markdown style notebooks, definitely use quarto's functions. I would still recommend binding molten's `EvaluateOperator` b/c this allows you to run arbitrary parts of a cell if you'd like.
2. Quarto gives you LSP features for code that's in a markdown doc. the python language server doesn't work for
````markdown
```python
# any code in here, will not normally have python LSP features
```
```` 
Quarto-nvim (otter.nvim really, but quarto handles setting up otter) does some magic to give you python lsp features when you're editing that python code.

Specifically regarding keymaps, I think that quarto is a "real" language server now, so it should use the same exact mappings that you setup for all the other language servers.

**Reply by Or1g3n** [reply]
Awesome, appreciate the clarification! I just got it up and running and working like a charm.
